quaistats/quaistats.go

```diff
@@ -924,18 +920,28 @@ func (s *Service) reportStats(conn *connWrapper) error {
 		cpuStat = []cpu.TimesStat{{}}
 	}
 
+	var ramUsed int64
+	var ramPercentUsed float32
 	// Get RAM usage
-	vmStat, err := mem.VirtualMemory()
-	if err != nil {
+	if vmStat, err := mem.VirtualMemory(); err == nil {
+		ramUsed = int64(vmStat.Used)
+		ramPercentUsed = float32(vmStat.UsedPercent)
+	} else {
 		log.Warn("Error getting RAM usage:", err)
-		vmStat = &mem.VirtualMemoryStat{UsedPercent: -1.0}
+		ramUsed = -1
```

Comment:
Ideally this (ram and swap) would be a constant errorvalue as well, seeing as we've had issues before with the error sentinel.

Reply:
Instead of having one for int and one for float, I would set it up as c_ramErrorValue and c_swapErrorValue and then cast it from an int (-1) to a float (-1.0) when you need.